src/stan/services/sample/hmc_nuts_diag_e_adapt.hpp

```diff
@@ -565,7 +553,7 @@ int hmc_nuts_diag_e_adapt(
     std::vector<InitWriter>& init_writer,
     std::vector<SampleWriter>& sample_writer,
     std::vector<DiagnosticWriter>& diagnostic_writer,
-    std::vector<callbacks::json_writer<Stream, Deleter>>& metric_writer) {
+    std::vector<callbacks::structured_writer>& metric_writer) {
```

Comment:
That error is because you didn't use a template inside this std::vector - you did the right thing for `hmc_nuts_dense_e_adapt` but it looks wrong in these other places

Reply:
thanks!